src/util/WebUtils.js

```diff
@@ -58,3 +57,44 @@ export const stringifyUriAndQuery = (uri: string, query: UriQueryMap): string =>
 export const omitLastChar = (uri: string, lastChar: string): string => {
   return uri.substr(-1) === lastChar ? uri.substr(0, uri.length - 1) : uri
 }
+
+/**
+ * -Replaces reserved characters with escape sequences representing the UTF-8
+ * encoding of the character.
+ * -Joins each query with '&'
+ */
+export const stringifyQuery = (query: UriQueryMap): string => {
+  return Object.keys(query)
+    .map(key => {
+      let out = encodeURIComponent(key)
+      if (query[key] != null) out += `=${encodeURIComponent(query[key])}`
+      return out
+    })
+    .join('&')
+}
+
+/**
+ * Parses the query portion of a URL/URI into a UriQueryMap.
+ * Does NOT extract the query from the complete URI!
+ * */
+export const parseQuery = (query?: string): UriQueryMap => {
```

Comment:
use the `url-parse` library already included in the GUI

Reply:
Added TODO